src/coreclr/vm/ceeload.cpp

```diff
@@ -4488,7 +4488,7 @@ Module::GetAssemblyIfLoaded(
             DomainAssembly * pCurAssemblyInExamineDomain = GetAssembly()->GetDomainAssembly();
             if (pCurAssemblyInExamineDomain == NULL)
             {
-                continue;
```

Comment:
My gut feeling, given the changes at about line 8450, is that this also used to be a useful loop.  The same consideration should likely be observed here.